modules/dxp/apps/commerce/headless/headless-commerce-machine-learning-impl/src/main/java/com/liferay/headless/commerce/machine/learning/internal/batch/v1_0/FrequentPatternRecommendationBatchEngineTaskItemDelegate.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ *
+ *
+ *
+ */
+
+package com.liferay.headless.commerce.machine.learning.internal.batch.v1_0;
+
+import com.liferay.batch.engine.BaseBatchEngineTaskItemDelegate;
+import com.liferay.batch.engine.BatchEngineTaskItemDelegate;
+import com.liferay.batch.engine.pagination.Page;
+import com.liferay.batch.engine.pagination.Pagination;
+import com.liferay.commerce.machine.learning.recommendation.FrequentPatternCommerceMLRecommendation;
+import com.liferay.commerce.machine.learning.recommendation.FrequentPatternCommerceMLRecommendationManager;
+import com.liferay.headless.commerce.machine.learning.dto.v1_0.FrequentPatternRecommendation;
+import com.liferay.portal.kernel.search.Sort;
+import com.liferay.portal.kernel.search.filter.Filter;
+import com.liferay.portal.kernel.util.ArrayUtil;
+
+import java.io.Serializable;
+
+import java.util.Map;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Riccardo Ferrari
+ */
+@Component(
+	enabled = false, immediate = true,
+	service = BatchEngineTaskItemDelegate.class
+)
+public class FrequentPatternRecommendationBatchEngineTaskItemDelegate
+	extends BaseBatchEngineTaskItemDelegate<FrequentPatternRecommendation> {
+
+	@Override
+	public void createItem(
+			FrequentPatternRecommendation item,
```

Comment:
@riccardo-ferrari I think we should name this `FrequentPatternRecommendation frequentPatternRecommendation ,`

Reply:
no objection. it is just consistency with all the other item delegates, I'll add a SF commit with all the batch item delegates renamed

Reply:
thanks!!